app/controllers/api/v1/searches_controller.rb

```diff
@@ -3,10 +3,18 @@ class Api::V1::SearchesController < Api::BaseController
   before_action :set_page, only: :show
 
   def show
-    @rubygems = Rubygem.legacy_search(params.require(:query)).with_versions.paginate(page: @page)
+    _, @rubygems = Rubygem.search(params.require(:query), es: es_enabled?, page: @page)
+    @rubygems = collect_source if @rubygems.is_a? Elasticsearch::Model::Response::Response
```

Comment:
Will `if es_enabled?` work here also?

Reply:
I removed the class check and moved logic of `collect_source` to `search` method itself. Please let me know if there is way I can avoid `? :`:
```
 @api ? result.map(&:_source) : [nil, result]
```